### PR TITLE
Omniorb

### PR DESCRIPTION
**Pull request by johanfforsberg**
OmniORB is a CORBA ORB implementation.

I am attempting to build conda packages for TANGO (http://tango-controls.org), and this is a dependency for it.

This is my first conda recipe and I do have one question: what to do about the "release" number? It's not really part of the version, but on the other hand, the conda package build number is also not the place to put it, right? I suppose it should not be important but still...

### CODE REVIEW

File: recipes/omniorb/build.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export CC=${PREFIX}/bin/gcc
```
Comment (patricksnape) [inline recipes/omniorb/build.sh]: I think you should use `toolchain` over fixing this - @jakirkham ?

**Comment by patricksnape** [comment]
What does the release number denote?

**Comment by johanfforsberg** [comment]
I'm not sure what the release number means (I'm talking about the final "2" in "4.2.1-2" in the name of tarball), I'm guessing that it's just for updates to the packaging, maybe makefiles and such. The version of the library is just 4.2.1 AFAICT and that should be all that matters here right?

File: recipes/omniorb/build.sh (hunk 1 above)
Comment (johanfforsberg) [inline recipes/omniorb/build.sh]: Do you mean adding "toolchain" as a build dependency?

**Comment by johanfforsberg** [comment]
Also, at this point, this package is only tested on linux. Seems like it should be possible to build omnniORB for windows and macos too, but I can't test those.

Is this a problem? Can I add something to the recipe to skip those builds? It looks like some automatic checks fail on windows.

File: recipes/omniorb/build.sh (hunk 1 above)
Comment (patricksnape) [inline recipes/omniorb/build.sh]: Yes.

File: recipes/omniorb/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  fn: omniORB-{{ version }}-{{ release }}.tar.gz
+  url: https://sourceforge.net/projects/omniorb/files/omniORB/omniORB-{{ version }}/omniORB-{{ version }}-{{ release }}.tar.bz2
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - gcc
```
Comment (patricksnape) [inline recipes/omniorb/meta.yaml]: I don't think you *need* the GCC compiler - so just omit it and the native compiler on each system will be used.

**Comment by patricksnape** [comment]
At the moment it is only passing on OSX! It seems to be failing on Linux due to not being able to find zlib. I think you can ignore that release number since it's probably just a packaging number (which we maintain by ourselves - `build: number`

File: recipes/omniorb/build.sh (hunk 1 above)
Comment (jakirkham) [inline recipes/omniorb/build.sh]: Yes, please drop this and the line below. They are not needed.

File: recipes/omniorb/meta.yaml (hunk 2 above)
Comment (jakirkham) [inline recipes/omniorb/meta.yaml]: Yep, please drop `gcc` and add `toolchain`.

File: recipes/omniorb/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    - zlib
```
Comment (jakirkham) [inline recipes/omniorb/meta.yaml]: Please pin to `1.2.*`. This is what we are using throughout the stack.

File: recipes/omniorb/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    - python
+  run:
+    - zlib
```
Comment (jakirkham) [inline recipes/omniorb/meta.yaml]: Please pin to `1.2.*`. This is what we are using throughout the stack.

File: recipes/omniorb/build.sh [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+export CXX=${PREFIX}/bin/g++
+
+
+./configure \
+  --prefix=$PREFIX \
+  --with-zlib=$PREFIX \
+
+make
+
```
Comment (jakirkham) [inline recipes/omniorb/build.sh]: Is there a `make check` we can run? If so, please add it.

File: recipes/omniorb/meta.yaml (hunk 2 above)
Comment (johanfforsberg) [inline recipes/omniorb/meta.yaml]: I don't get think I get it; if I do that, configure does not find a compiler. 

I'm trying to build on a CentOS 7 machine by the way. Is toolchain something that will only work on conda-forge?

**Comment by johanfforsberg** [comment]
The only way I've found for fixing the zlib issue is something like this in the build script:
```
export CXXFLAGS="-I${PREFIX}/include" 
```
Would that be OK? I believe that flag is picked up by configure so it's not compiler specific.

**Comment by jakirkham** [comment]
Have you tried adding `pkg-config`?

**Comment by johanfforsberg** [comment]
Adding pkg-config as a build dep did not help.

**Comment by jakirkham** [comment]
That's unfortunate. Yeah, I don't see any problem with adding the includes to `CXXFLAGS`. Though I would do it like this `export CXXFLAGS="${CXXFLAGS} -I${PREFIX}/include" `.

**Comment by johanfforsberg** [comment]
As far as I can tell, the windows build is still broken. It should be possible to build omniorb for windows but I have no environment to test this in. It requires cygwin and visual studio. Any advice on this?

Also I still haven't figured out the issue with gcc vs. toolchain.

File: recipes/omniorb/meta.yaml (hunk 2 above)
Comment (patricksnape) [inline recipes/omniorb/meta.yaml]: It sets a load of compiler flags so it should work - however, it was designed to be used with the CircleCI/CentOS 6 docker instances.

**Comment by patricksnape** [comment]
The Python ecosystem requires building certain versions of Python with certain version of Visual Studio and mingw is not currently a supported build platform. Why do you require 'cygwin' to build on Windows? Because of the build system?

**Comment by johanfforsberg** [comment]
I don't know why cygwin is needed, I'm just looking at the build requirements for OmniORB. I guess the build scripts use some tools from there, e.g. "make".

For my part it would be fine to ignore the windows build for now, since we're not using windows anyway. I guess someone who cares about it can come along and fix it later.

File: recipes/omniorb/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  name: omniorb
+  version: {{ version }}
+
+source:
+  fn: omniORB-{{ version }}-{{ release }}.tar.gz
+  url: https://sourceforge.net/projects/omniorb/files/omniORB/omniORB-{{ version }}/omniORB-{{ version }}-{{ release }}.tar.bz2
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
```
Comment (patricksnape) [inline recipes/omniorb/meta.yaml]: To skip Windows just add `skip: True  # [win]` under the `build` section.

File: recipes/omniorb/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  skip: True  # [win]
+  
+requirements:
+  build:
+    - gcc
```
Comment (jakirkham) [inline recipes/omniorb/meta.yaml]: Please replace with `toolchain`.

**Comment by patricksnape** [comment]
That's a strange error - may need to `cat` that `config.log` to see what was reported.

**Comment by johanfforsberg** [comment]
Sorry for not responding for a while. 

From config.log:
```
configure:3169: checking whether the C compiler works
configure:3191: /home/johfor/anaconda3/conda-bld/omniorb_1476104326481/_b_env_placehold_placehol/bin/gcc  -m64   conftest.c
  >&5
./configure: line 3193: /home/johfor/anaconda3/conda-bld/omniorb_1476104326481/_b_env_placehold_placehol/bin/gcc: No such f
ile or directory
configure:3195: $? = 127
configure:3233: result: no
```
Could it be that the configure script is somehow hardcoded to assume that gcc will be the compiler?

**Comment by patricksnape** [comment]
Looks like it!

**Comment by johanfforsberg** [comment]
Is there some documentation on how the toolchain package works? As far as I can tell, it sets a bunch of environment vars to tell configure/make what compiler to use and what flags depending on platform. But I don't understand how it provides the compiler itself.

**Comment by bilderbuchi** [comment]
I have found that there is an OmniORB package available on https://anaconda.org/mutirri/omniorb/files. It even contains a build for Windows. I am not familiar with the build/recipe process, but if you download [one of the windows bulds](https://anaconda.org/mutirri/omniorb/4.2.1/download/win-64/omniorb-4.2.1-py27_0.tar.bz2), there is an info folder with some recipe-related stuff (including a small patch for Windows). So, maybe one could dissect those files to arrive at a working build process for all platforms?

**Comment by bilderbuchi** [comment]
Specifically, I found in the `meta.yaml` a way to set a windows-only patch:
```
source:
    fn: omniORB-4.2.1-2.tar.bz2
    url: https://sourceforge.net/projects/omniorb/files/omniORB/omniORB-4.2.1/omniORB-4.2.1-2.tar.bz2

    patches:
        - 0001-Prepared-building-environment-for-windows-platform.patch  # [win]
```
with `0001-Prepared-building-environment-for-windows-platform.patch` containing:
```
From b23c7bf62da7f70b1e082a851292746d316db776 Mon Sep 17 00:00:00 2001
From: "Kamil Kwiek (irritum)" <kamil.kwiek@continuum.io>
Date: Mon, 21 Mar 2016 11:53:35 +0100
Subject: [PATCH] Prepared building environment for windows platform

---
 config/config.mk               | 2 +-
 mk/platforms/x86_win32_vs_8.mk | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git config/config.mk config/config.mk
index 5567123..a4d4495 100644
--- config/config.mk
+++ config/config.mk
@@ -65,7 +65,7 @@
 #platform = x86_win32_vs_11
 #platform = x86_win32_vs_10
-#platform = x86_win32_vs_9
+platform = x86_win32_vs_9
 #platform = x86_win32_vs_8
 #platform = x86_win32_vs_7
 #platform = x86_win32_vs_6
 #platform = x86_win32_mingw
diff --git mk/platforms/x86_win32_vs_9.mk mk/platforms/x86_win32_vs_9.mk
index be80cf9..b91d1a9 100755
--- mk/platforms/x86_win32_vs_9.mk
+++ mk/platforms/x86_win32_vs_9.mk
@@ -23,7 +23,7 @@ ABSTOP = $(shell cd $(TOP); pwd)
 # http://sourceforge.net/project/showfiles.php?group_id=51138&package_id=48638
 # In that case, uncomment the first line below.
 
-#PYTHON = $(ABSTOP)/$(BINDIR)/omnipython
+PYTHON = python
 #PYTHON = /cygdrive/c/Python26/python
 
 
-- 
2.0.5
```

Btw Python 3 compatibility would be ace, cause a next step would be to get omniORBpy packaged to et on a path to Python 3 compatibility for OMPYthon (OpenModelica/OMPython#1).

**Comment by carlodri** [comment]
@conda-forge/staged-recipes this can be closed as superseded by #7835 (the original author confirmed there that he is no more interested in this PR)

**Comment by scopatz** [comment]
This was an old one :)

**Comment by carlodri** [comment]
oldies but goodies 😉

**Pull request closed without merging**